docs/Guides/creating_new_vendor_app.md

```diff
@@ -0,0 +1,137 @@
+
+The next steps are for doing that.
+
+### Step 5 - Tagging the account group id or the partner id to the auth url
+
+Lets add a custom AuthorizationRequestResolver to our authorization endpoint like so
+
+![image.png](../../assets/images/guides/vendor_application/authorization_endpoint.png)
+
+
+We have added the authorizationEndpoint to override the default behavior of our auth endpoints. This comes with a handy authorizationRequestresolver option to use a CustomRequestResolver.
+
+We use a parameterized constructor to send both the clientRegistrationRepository(Ideally stuff from your application.yml file) and the authorizationUrl that we want to override. So lets go ahead and create our class
+
+![image.png](../../assets/images/guides/vendor_application/authorization_request_resolver.png)
+
+
+
+Let's create a class variable called defaultRequestResolver of type Oauth2AuthorizationRequestResolver to get the default configurations and during the constructor initialization, let's initialize this variable with the clientRegistrationRepo for vendasta and the auth url which was hardcoded from the previous step.
+
+The class should implement the Oauth2AuthorizationRequestResolver so that means we need to implement the resolve methods. But since we are only interested in integrating Vendasta as the provider let's use the same logic for both the overridden overloaded methods(That’s a mouthful huh!! Welcome to java) . 
+
+Let's create a method `createCustomEntry` to take care of this activity.
+
+![image.png](../../assets/images/guides/vendor_application/create_custom_entry.png)
+
+
+The method does a lot of things and it could be engineered better, but to make it screenshot friendly, we have made everything available here. Let's look at what’s happening here
+
+1. Let's call the resolve method of the default implementation of this class by spring boot . (We need this because we need various configuration setup like adding the clientRegistrationId,state,redirectUrl and such. lets create a temp Oauth2AuthorizationRequest.
+
+2. The second line  gets the accountId from the request Url(This is a mandatory step when integrating with Vendasta. When creating the entry Url , The accountId should be tagged in the format `/entry/accountId` for it to work properly. This is explained better in the docs above.
+
+3. The next if statement will make sure that we redirect the user to the right location after the login. This also works hand in hand with another problem. That is to look for an active session with the client. If there is no active session with the client then this variable is null. So the auth flow happens only when there is no valid sessions. 
+
+4. The next line sets the accountId in the `state` parameter of your authUrl.
```

Comment:
We do suggest that certain data needed down the line(e.g. app_id if using multiple apps sharing the same entry path, or the nextUrl) be passed through sso via the state param, but [you still need it to be random](https://auth0.com/docs/secure/attack-protection/state-parameters). So likely would want to append the account_id to something random rather than have it stand alone. This guide is so close to being comprehensive, may as well go all the way.